Common/src/main/java/at/petrak/hexcasting/common/casting/actions/spells/OpIgnite.kt

```diff
@@ -30,27 +27,22 @@ object OpIgnite : SpellAction {
             args: List<Iota>,
             env: CastingEnvironment
     ): SpellAction.Result {
-        when (val target = args[0]) {
-            is EntityIota -> {
-                val entity = args.getEntity(0, argc)
-                env.assertEntityInRange(entity)
-                return SpellAction.Result(
-                    EntitySpell(entity),
-                    MediaConstants.DUST_UNIT,
-                    listOf(ParticleSpray.burst(entity.position(), 1.0))
-                )
-            }
-            is Vec3Iota -> {
-                val block = args.getBlockPos(0, argc)
-                env.assertPosInRangeForEditing(block)
-                return SpellAction.Result(
-                    BlockSpell(block),
-                    MediaConstants.DUST_UNIT,
-                    listOf(ParticleSpray.burst(Vec3.atCenterOf(BlockPos(block)), 1.0))
-                )
-            }
-            else -> throw MishapInvalidIota.ofType(target, 0, "entity_or_vector")
-        }
+        val target = args.getEntityOrBlockPos(0, argc)
+
+        target.map(env::assertEntityInRange, env::assertPosInRange)
+
+        val burstPos = target.map({ent -> if (ent is ItemEntity) {
+            // Special case these because the render is way above the entity
+            ent.position().add(0.0, 3.0 / 8.0, 0.0)
```

Comment:
maybe this logic could be moved to a utility function? eg `ParticleSpray.getEntityBurstPos` or something